neo4j fails to start

```
2024-07-30 13:57:56.118+0000 ERROR Failed to start Neo4j on 0.0.0.0:7474.
java.lang.RuntimeException: Error starting Neo4j database server at /data/databases
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:227) ~[neo4j-4.4.33.jar:4.4.33]
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.build(DatabaseManagementServiceFactory.java:180) ~[neo4j-4.4.33.jar:4.4.33]
        at org.neo4j.server.CommunityBootstrapper.createNeo(CommunityBootstrapper.java:36) ~[neo4j-4.4.33.jar:4.4.33]
        at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:142) [neo4j-4.4.33.jar:4.4.33]
        at org.neo4j.server.NeoBootstrapper.start(NeoBootstrapper.java:95) [neo4j-4.4.33.jar:4.4.33]
        at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:34) [neo4j-4.4.33.jar:4.4.33]
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.procedure.impl.GlobalProceduresRegistry@1a2e0d57' was successfully initialized, but failed to start. Please see the attached cause exception "Some jar procedure files (apoc.jar) are invalid, see log for details.".
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:463) ~[neo4j-common-4.4.33.jar:4.4.33]
        at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.4.33.jar:4.4.33]
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:218) ~[neo4j-4.4.33.jar:4.4.33]
        ... 5 more
Caused by: java.util.zip.ZipException: Some jar procedure files (apoc.jar) are invalid, see log for details.
        at org.neo4j.procedure.impl.ProcedureJarLoader.loadProceduresFromDir(ProcedureJarLoader.java:83) ~[neo4j-procedure-4.4.33.jar:4.4.33]
        at org.neo4j.procedure.impl.GlobalProceduresRegistry.start(GlobalProceduresRegistry.java:371) ~[neo4j-procedure-4.4.33.jar:4.4.33]
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:442) ~[neo4j-common-4.4.33.jar:4.4.33]
        at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:110) ~[neo4j-common-4.4.33.jar:4.4.33]
        at org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.startDatabaseServer(DatabaseManagementServiceFactory.java:218) ~[neo4j-4.4.33.jar:4.4.33]
        ... 5 more
```        

The reason is that the apoc.jar file was downloaded with an error.
```bash
netshoot-pod:/mnt/data# ls -ltrh
total 57M    
-rw-r--r--    1 7474     7474       10.0M Jul 30 13:53 apoc.jar
-rw-r--r--    1 7474     7474       19.5M Jul 30 13:53 apoc.jar.1
-rw-r--r--    1 7474     7474       27.0M Jul 30 13:53 streams.jar
netshoot-pod:/mnt/data# md5sum *
f3f00e6d2178a2cc34ec97faf66d7880  apoc.jar
2597257c3872e0411e9c831285a0b4ce  apoc.jar.1
2ef94f0744503050f3ff91b85e7e78c5  streams.jar
```

--- a/crossplane/xrd_kcl_function.k
+++ b/crossplane/xrd_kcl_function.k
@@ -194,6 +194,8 @@ items = [
                     workingDir = "/var/lib/neo4j/plugins"
                     command = [
                         "curl",
+                        "--retry"
+                        "3"
                         "-Lo"
                         "apoc.jar"
                         "--no-clobber"
@@ -211,6 +213,8 @@ items = [
                     }]
                     command = [
                         "curl",
+                        "--retry"
+                        "3"
                         "-Lo"
                         "streams.jar"
                         "--no-clobber"
@@ -236,7 +240,7 @@ items = [
         }
         namespace = makeContextNamespace("{{ context_name }}")
         values = {
-            image = { 
+            image = {
                 # TODO(@metacoma) parametrize
                 repository = "${config.vector.image.repo}"
                 tag = "${config.vector.image.tag}"
@@ -247,14 +251,14 @@ items = [
                 transforms = {
                     cloudevent = {
                         type = "lua"
-                        version = "2" 
-                        inputs = ["neo4j-cdc"] 
+                        version = "2"
+                        inputs = ["neo4j-cdc"]
                         hooks = {
                             process = "process"
-                            init = "seed_init" 
-                        } 
+                            init = "seed_init"
+                        }
                         source = '''\
-        function seed_init() 
+        function seed_init()
           uuid = require("uuid")
           local socket = require("socket")  -- gettime() has higher precision than os.time()
           uuid.seed()
@@ -292,12 +296,12 @@ items = [
           emit(cloudevent)
         end
 '''
-                    } 
-                } 
+                    }
+                }
                 sources = {
                     "neo4j-cdc" = {
                         type = "kafka"
-                        bootstrap_servers: "${config.kafka.brokerEndpoint}" 
+                        bootstrap_servers: "${config.kafka.brokerEndpoint}"
                         group_id = "{{ context_name }}-vector"
                         topics = [ "{{ context_name }}-cdc-topic" ]
                         decoding.codec = "json"
@@ -314,14 +318,14 @@ items = [
                             "fetch.min.bytes" = "1000000"
                         }
 
-                    } 
-                } 
+                    }
+                }
                 sinks = {
                     console = {
                         type = "console"
                         inputs = [ "neo4j-cdc" ]
                         encoding.codec = "json"
-                    } 
+                    }
                     "context-broker" = {
                         type = "http"
                         inputs = [ "cloudevent" ]
@@ -330,7 +334,7 @@ items = [
                         batch = {
                             max_size = 1
                             max_events = 1
-                        } 
+                        }
                         framing.method = "newline_delimited"
                         uri = "http://broker-ingress.${config.knative.eventing.namespace}.svc.cluster.local/" +  makeContextNamespace("{{ context_name }}") + "/" + contextBroker.name("{{ context_name }}")
                         request = {
@@ -339,9 +343,9 @@ items = [
                                 "Content-Type" = "application/cloudevents+json"
                             }
                         }
-                    } 
-                } 
-            } 
+                    }
+                }
+            }
         }
       }
     }
@@ -497,7 +501,7 @@ kind: InMemoryChannel
                 }
             }
             filters = [
-                { 
+                {
                     "any" = [
                         { exact.type = "iodocument" }
                     ]
@@ -716,7 +720,7 @@ kind: InMemoryChannel
                 }
             }
             filters = [
-                { 
+                {
                     "any" = [
                         { exact.type = "iodocument" }
                         { exact.type = "touch" }
@@ -838,7 +842,7 @@ spec:
                 }
             }
             filters = [
-                { 
+                {
                     "any" = [
                         { exact.type = "iodocument" }
                         { exact.type = "touch" }

